yt/frontends/ramses/data_structures.py

```diff
@@ -591,28 +678,33 @@ def __init__(
             max_level, max_level_convention
         )
 
-        # Infer if the output is organized in groups
-        root_folder, group_folder = os.path.split(os.path.split(filename)[0])
+        file_handler = RAMSESFileSanitizer(filename)
+
+        # This should not happen, but let's check nonetheless.
+        if not file_handler.is_valid:
+            raise ValueError(
+                "Invalid filename found when building a RAMSESDataset object: %s",
+                filename,
+            )
+
+        # Sanitize the filename
+        info_fname = file_handler.info_fname
 
-        if group_folder == "group_00001":
-            # Count the number of groups
-            # note: we exclude the unlikely event that one of the group is actually a
-            # file instad of a folder
+        if file_handler.group_name is not None:
             self.num_groups = len(
                 [
                     _
-                    for _ in glob(os.path.join(root_folder, "group_?????"))
+                    for _ in glob(str(file_handler.root_folder / "group_?????"))
                     if os.path.isdir(_)
```

Comment:
I'm down for pathlib anytime, just don't mix it with os.path:
```suggestion
                    for _ in file_handler.root_folder.glob("group_?????")
                    if _.is_dir()
```